src/model/documentselection.js

```diff
@@ -276,12 +261,17 @@ export default class DocumentSelection {
 	 *		</blockB>
 	 *		<blockE></blockE>]
 	 *
-	 * **Note:** To get all selected blocks use {@link #getSelectedBlocks `getSelectedBlocks()`}.
+	 * **Special case**: If a selection ends at the beginning of a block, that block is not returned as from user perspective
+	 * this block wasn't selected. See [#984](https://github.com/ckeditor/ckeditor5-engine/issues/984) for more details.
+	 *
+	 *		<paragraph>[a</paragraph>
+	 *		<paragraph>b</paragraph>
+	 *		<paragraph>]c</paragraph> // this block will not be returned
```

Comment:
If I understood the code correctly, in following example:

```
<block>
  <blockA>[a</blockA>
  <blockB>b]</blockB>
</block>
```

Blocks A and B will be returned? It makes obvious sense, so I'd add it to examples. However what is returned in this scenario:

```
<blockX>
  <blockA>[a</blockA>
  <blockB>b</blockB>
</blockX>
<blockY>y]</blockY>
```

A, B, Y?

Reply:
probably the `a,b,y` but I haven't tested this case. At the moment we do not have such structures. Taking tables as an example of a block that nests other blocks - the `blockX` would be a limit also so that selection wouldn't be possible and be fixed to `[<blockX>...y]</blockY>`

Reply:
I've moved the conversion on this below.